devtools/x-core/src/lib.rs

```diff
@@ -56,7 +56,11 @@ impl XCoreContext {
             config,
             current_dir,
             current_rel_dir,
-            git_cli: GitCli::new(project_root)?,
+            git_cli: if let Ok(cli) = GitCli::new(project_root) {
+                Some(cli)
+            } else {
+                None
+            },
```

Comment:
You could even make this use a `OnceCell` initialized lazily if you're feeling fancy.

Reply:
ah, saw this afterwards, second fixup does this.